azurerm/internal/services/sql/helper/sqlExtendedAuditing.go

```diff
@@ -0,0 +1,103 @@
+package helper
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/2017-03-01-preview/sql"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func BlobExtendedAuditingSchemaFrom(s map[string]*schema.Schema) map[string]*schema.Schema {
+	blobAuditing := map[string]*schema.Schema{
+		"blob_extended_auditing_policy": {
+			Type:     schema.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"storage_account_access_key": {
+						Type:         schema.TypeString,
+						Required:     true,
+						Sensitive:    true,
+						ValidateFunc: validate.NoEmptyStrings,
+					},
+
+					"storage_endpoint": {
+						Type:         schema.TypeString,
+						Required:     true,
+						ValidateFunc: validate.URLIsHTTPS,
+					},
+
+					"storage_secondary_key_enabled": {
```

Comment:
Looking into this more, i think `storage_account_access_key_is_secondary` is the most accurate name?
```suggestion
					"storage_account_access_key_is_secondary": {
```